monitor/main.py

```
@@ -220,6 +249,63 @@ def offline_logger(self, validator, steps):
         with open(self.report_dir / filename, "w") as f:
             json.dump({"validator": encode_hex(validator), "missed_steps": steps}, f)
 
+    def equivocation_logger(self, equivocated_block_hashes):
+        """Log a reported equivocation event.
+
+        Each new equivocation report is logged into a new file, using a counter
+        as unique identifier.
+        Logged information are the proposer of the blocks, the height at which
+        all blocks have been equivocated and a list of all block hashes with
+        their timestamp.
+        Additionally two representing blocks are logged with their RLP encoded
+        header and related signature, which can be used for an equivocation
+        proof on reporting a validator.
+        """
+
+        blocks = [
+            self.w3.eth.getBlock(block_hash) for block_hash in equivocated_block_hashes
+        ]
+
+        block_hashes_and_timestamp_strings = [
+            BLOCK_HASH_AND_TIMESTAMP_TEMPLATE.format(
+                block_hash=encode_hex(block.hash),
+                block_timestamp=datetime.datetime.utcfromtimestamp(block.timestamp),
+            )
+            for block in blocks
+        ]
+
+        block_hash_and_timestamp_summary = "\n".join(block_hashes_and_timestamp_strings)
+
+        # Use the first two blocks as representational data for the equivocation proof.
+        block_one = get_canonicalized_block(blocks[0])
+        block_two = get_canonicalized_block(blocks[1])
```

Maybe add a 
`assert len(equivocated_block_hashes) >=2 `
at the beginning of this function to fail early if used wrong

---

Should I make a comment for the purpose or is it self-explaining?

---

self explaining i think